odoo/addons/base/models/res_company.py

```diff
@@ -198,37 +198,41 @@ def _company_default_get(self, object=False, field=False):
     def cache_restart(self):
         self.clear_caches()
 
-    @api.model
-    def create(self, vals):
-        if not vals.get('favicon'):
-            vals['favicon'] = self._get_default_favicon()
-        if not vals.get('name') or vals.get('partner_id'):
+    @api.model_create_multi
+    def create(self, vals_list):
+        companies = self.browse()
+        for vals in vals_list:
+            if not vals.get('favicon'):
+                vals['favicon'] = self._get_default_favicon()
+            if not vals.get('name') or vals.get('partner_id'):
+                self.clear_caches()
+                companies |= super().create(vals)
```

Comment:
Doing an "or" does not guarantee the record set order ... would use a "+" to avoid unwanted side effects.

Reply:
FYI the `|` operator on recordsets is ordered: it returns the concatenation where second occurrences of elements have been removed.

Reply:
Ah ok, thought it was not the case, thanks for the info :) . Anyway even if ordered I prefer the + that is explicit when there is no issue with duplicates (as we are in a create we will not have duplicates).